README.md

```diff
@@ -1,87 +1,158 @@
+    * To indicate where solidity should find libraries, you provide an [import remapping][remapping] to `solc` using the `--solc-remapping` argument.  For example:
+      ```bash
+      cargo gambit path/to/file.sol \
+        --solc-remapping @openzepplin=node_modules/@openzeppelin \
+        --solc-remapping ...
+      ```
+    * To include additional allowed paths,
+      you provide solidity's [allowed paths][allowed] to `solc` using the `--allow-paths` argument.
+      For example:
+      ```bash
+      cargo gambit path/to/file.sol --solc-allowpaths @openzepplin=... --solc-allowpaths ...
+      ```
+
+[remapping]: https://docs.soliditylang.org/en/v0.8.17/path-resolution.html#import-remapping
+[basepath]: https://docs.soliditylang.org/en/v0.8.17/path-resolution.html#base-path-and-include-paths
+[allowed]: https://docs.soliditylang.org/en/v0.8.17/path-resolution.html#allowed-paths
+
+#### Running Gambit Through a Configuration File
+This is the recommended way to run Gambit.
+This approach allows you to control and localize
+mutation generation and is easier
+to use than passing many command line flags.
+
+To run gambit with a configuration file, simply pass the name of the `json` file:
+```bash
+cargo gambit-cfg benchmarks/config-jsons/test1.json
 ```
+
+The configuration file is a [json][json-spec] file containing the command line
+arguments for `gambit` and additional configuration options.
+For example, the following configuration is equivalent
+to `gambit benchmarks/10Power/TenPower.sol --solc-remapping @openzepplin=node_modules/@openzeppelin`:
+
+```json
 {
-    "filename": "path/to/file.sol",
+    "filename": "benchmarks/10Power/TenPower.sol",
     "remappings": [
-        "@openzeppelin=PATH/TO/node_modules/@openzeppelin"
+        "@openzeppelin=node_modules/@openzeppelin"
     ]
 }
 ```
-or
+
+In addition to the specifying the command line arguments, you can list the
+specific {ref}`types of mutations <mutation-types>` that you want to apply, the
+specific functions you wish to mutate, and more.  See {ref}`gambit-config` for
+more details, and [the `benchmark/config-jsons` directory][config-examples] for
+examples.
+
+[json-spec]: https://json.org/
+[config-examples]: https://github.com/Certora/gambit/blob/master/benchmarks/config-jsons/
+[test6]: https://github.com/Certora/gambit/blob/master/benchmarks/config-jsons/test6.json
+
+
+#### Configuring the Set of Mutations, Functions, and Contracts
+If you are using Gambit through a configuration file,
+you can localize the mutations to some
+functions and contracts.
+You can also choose which mutations you want (see {ref}`mutation-types` for the list of possible mutations).
+Here is an example that shows how to configure these options.
 ```
-{
-    "filename": "path/to/file.sol",
-    "solc-allowpaths": [
-        "path1",
-        "path2"
-    ]
-}
+[
+    {
+        "filename": "Foo.sol",
+        "contract": "C",
+        "functions": ["bar", "baz"],
+        "solc": "solc5.12"
```

Comment:
Can we use a different solc between the two files in this example?
Also, maybe we can use a more modern version for the first example so that the tool appears up to date

Reply:
`solc5.12` is ancient - it came out in 2019